api4/bot_test.go

```diff
@@ -713,16 +713,116 @@ func TestDisableBot(t *testing.T) {
 		CheckCreatedStatus(t, resp)
 		defer th.App.PermanentDeleteBot(bot.UserId)
 
-		disabledBot1, resp := th.Client.DisableBot(bot.UserId)
+		enabledBot1, resp := th.Client.DisableBot(bot.UserId)
 		CheckOKStatus(t, resp)
-		bot.UpdateAt = disabledBot1.UpdateAt
-		bot.DeleteAt = disabledBot1.DeleteAt
-		require.Equal(t, bot, disabledBot1)
+		bot.UpdateAt = enabledBot1.UpdateAt
+		bot.DeleteAt = enabledBot1.DeleteAt
+		require.Equal(t, bot, enabledBot1)
+
+		// Check bot disabled
+		disab, resp := th.SystemAdminClient.GetBotIncludeDeleted(bot.UserId, "")
+		CheckOKStatus(t, resp)
+		require.NotZero(t, disab.DeleteAt)
+
+		// Disabling should be idempotent.
+		enabledBot2, resp := th.Client.DisableBot(bot.UserId)
+		CheckOKStatus(t, resp)
+		require.Equal(t, bot, enabledBot2)
+	})
+}
+
+func TestEnableBot(t *testing.T) {
+	t.Run("enable non-existent bot", func(t *testing.T) {
```

Comment:
Looks like here is some repeated code, maybe we can use a test table here.

Reply:
IMO, this is one of those cases where the table test would have so many flags to control the variability of the test, it would obscure what's under test in the first place. Agree that table testing is great for repetitive test coverage, though!

Reply:
Ya going to agree with Jesse here. Converting this into a table test is probably more work then it's worth.